### PR TITLE
[Form] Compare fields inside a RepeatedType through compare option

### PR DESCRIPTION
**Pull request by MetalArend**
This adds documentation on the compare field, added by a new feature PR on symfony/symfony.

### CODE REVIEW

File: reference/forms/types/repeated.rst [hunk 1]
```diff
@@ -177,6 +178,21 @@ type
...
 ``PasswordType::class`` will render two password fields.
 
+compare
+~~~~~~~
+
+**compare**: ``callable`` **default**: ``null``
+
+The two underlying fields' values will be compared by this callable. The callable receives two values to compare, and should return a boolean: true if the fields are considered equal, false otherwise.
+
+When this option has a null value, it uses internally a callable comparable to:
```
Comment (xabbuh) [inline reference/forms/types/repeated.rst]: You need a second colon here so that the following paragraph is treated as a code block.

Comment (MetalArend) [inline reference/forms/types/repeated.rst]: So, might be more interesting to add backticks around that code block?

Comment (xabbuh) [inline reference/forms/types/repeated.rst]: No not really. The alternative to the double colon would be the `code-block` directive like this:

```rst
.. code-block:: php

    $builder->add('password', RepeatedType::class, array(
        'compare'  => function ($value1, $value2) {
            return $value1 === $value2;
        },
    ));
```

But throughout the docs we prefer the double colon if the previous paragraph ended with a colon and we want to show a PHP code example.

File: reference/forms/types/repeated.rst [hunk 2]
```diff
@@ -18,6 +18,7 @@ accuracy.
 |             | - `second_name`_                                                       |
 |             | - `second_options`_                                                    |
 |             | - `type`_                                                              |
+|             | - `compare`_                                                           |
```
Comment (HeahDude) [inline reference/forms/types/repeated.rst]: Form options are sorted alphabetically, can you please preserve it?

File: reference/forms/types/repeated.rst [hunk 3]
```diff
@@ -177,6 +178,21 @@ type
 The two underlying fields will be of this field type. For example, passing
 ``PasswordType::class`` will render two password fields.
 
+compare
+~~~~~~~
+
+**compare**: ``callable`` **default**: ``null``
```
Comment (HeahDude) [inline reference/forms/types/repeated.rst]: ``**compare**` => `**type**`

File: reference/forms/types/repeated.rst [hunk 4]
```diff
@@ -177,6 +178,21 @@ type
...
+
+The two underlying fields' values will be compared by this callable. The callable receives two values to compare, and should return a boolean: true if the fields are considered equal, false otherwise.
```
Comment (HeahDude) [inline reference/forms/types/repeated.rst]: I think you can omit the first sentence and start directly with `The callable receives two values to compare`.

File: reference/forms/types/repeated.rst [hunk 5]
```diff
@@ -177,6 +178,21 @@ type
...
+
+When this option has a null value, it uses internally a callable comparable to::
```
Comment (HeahDude) [inline reference/forms/types/repeated.rst]: "By default a strict comparison is done."  or something similar could be enough.

Instead you could give an example (not necessarily with code) where the option can be useful.

**Review by HeahDude — CHANGES_REQUESTED** [review]

**Review by HeahDude — APPROVED** [review]
Very nice improvement to use the key as third argument 👍

**Comment by MetalArend** [comment]
Should I rebase this on master, to be able to complete the Travis CI build?

**Comment by xabbuh** [comment]
@MetalArend I restarted the build. It's passing now.

File: reference/forms/types/repeated.rst [hunk 6]
```diff
@@ -114,6 +115,15 @@ be displayed when the two fields do not match each other.
 Field Options
 -------------
 
+compare
+~~~~~~~
+
+**type**: ``callable`` **default**: ``null``
+
+The callable receives two values to compare, and should return a boolean: true if the fields are considered equal, false otherwise. The callable also receives the currently compared key as a third value.
```
Comment (HeahDude) [inline reference/forms/types/repeated.rst]: Is it worth having "try", "false", and function like "password_hash" surrounded with back sticks?

**Pull request closed without merging**